packages/app/src/vis-packs/core/heatmap/MappedHeatmapVis.tsx

```diff
@@ -58,22 +65,39 @@ function MappedHeatmapVis(props: Props) {
     customDomain,
     colorMap,
     scaleType,
+    complexVisType,
     keepRatio,
     showGrid,
     invertColorMap,
     flipXAxis,
     flipYAxis,
   } = config;
 
-  const numArray = useToNumArray(value);
+  const { shape: dims, type } = dataset;
+  const isComplex = isComplexType(type);
+
+  // If complex dataset displayed as phase/amplitude: `numArray` = amplitude; `alphaArray` = phase
+  const isPhaseAmp = complexVisType === ComplexVisType.PhaseAmplitude;
+
+  const numArray = useToNumArray(
+    value,
+    isPhaseAmp ? ComplexVisType.Amplitude : complexVisType,
+  );
+  const phaseArray = useToNumArray(
+    isComplex && isPhaseAmp ? value : undefined,
+    ComplexVisType.Phase,
+  );
+
   const numAxisArrays = useToNumArrays(axisValues);
 
-  const { shape: dims } = dataset;
   const [slicedDims, slicedMapping] = useSlicedDimsAndMapping(dims, dimMapping);
   const dataArray = useMappedArray(numArray, slicedDims, slicedMapping);
+  const alphaArray = useMappedArray(phaseArray, slicedDims, slicedMapping);
 
   const dataDomain =
     useDomain(dataArray, scaleType, undefined, ignoreValue) || DEFAULT_DOMAIN;
+  const alphaDomain = useDomain(alphaArray, scaleType) || DEFAULT_DOMAIN;
```

Comment:
`alphaArray` can be `undefined` so I had to account for that in `useDomain` (and therefore `getBounds`).